packages/pn-validator/src/Validator.ts

```diff
@@ -4,12 +4,39 @@ import { ValidatorBuilder } from './ValidatorBuilder';
 import { hasError } from './HasError';
 import { TypeRules } from './types/TypeRules';
 
+/**
+ * Validator options that include the strict mode but in the future you can add any other
+ * options.
+ */
+export interface ValidatorOptions { 
+  strict?: boolean;
+}
+
 export class Validator<TModel> {
   private validatorBuilders: ValidatorBuilders<TModel> = {};
+  private strict: boolean; // strict mode
+
+  constructor(options: ValidatorOptions = {}) {
+    this.strict = options.strict ?? false;
+  }
 
   public readonly validate = (model: TModel): ValidationError<TModel> | null => {
+
+    // check if the number of rule keys are equals to the number of model keys.
+    if (this.strict) {
+      if (typeof model === 'object' && model !== null) {
+        const modelKeys = Object.keys(model);
+        const ruleKeys = Object.keys(this.validatorBuilders);
+        const missingKeys = modelKeys.filter((key) => !ruleKeys.includes(key));
```

Comment:
This check isn't complete. The model can have a hierarchical structure; we can have a property that is an object and we can have a property that is an array of object. All those properties must be checked.

Check the updates on the issue